moo.js

```diff
@@ -437,7 +432,7 @@
       var s = this.states[key]
       map[key] = {
         groups: s.groups,
-        regexp: new RegExp(s.regexp.source, s.regexp.flags),
+        regexp: s.regexp,
         error: s.error,
       }
     }
```

Comment:
You should actually be able to get rid of all the state-map copying code here, and just hand `this.states` to the new Lexer.